modules/dex/src/lib.rs

```diff
@@ -1176,6 +1174,17 @@ impl<T: Config> Pallet<T> {
 		Ok(supply_amounts)
 	}
 
+	fn validate_path(path: &[CurrencyId]) -> DispatchResult {
+		let path_length = path.len();
+		ensure!(
+			path_length >= 2 && path_length <= T::TradingPathLimit::get().saturated_into(),
+			Error::<T>::InvalidTradingPathLength
+		);
+		ensure!(path[0] != path[path_length - 1], Error::<T>::InvalidTradingPath);
```

Comment:
```suggestion
		ensure!(path.iter().unique().len() == path_length, Error::<T>::InvalidTradingPath);
```

and make it `public` for the `validate_path`

Reply:
Checking unique is much worse than getting elements & compare in performance.

Reply:
I think `kUSD, KAR, KAR, KSM` and `KUSD, KSM, KUSD, KAR` should be checked.

Reply:
It's common to trade across two pools. I don't see the benefits of disabling them.

Reply:
`KUSD, KSM, KSM, KAR` will generate the tradingpair `KUSD, KSM`, `KSM, KSM`,  `KSM,KAR`. But the `KSM, KSM` is invalid.
`KUSD, KSM, KUSD, KAR` should be `KUSD, KAR`.